src/Illuminate/Mail/Mailable.php

```diff
@@ -122,15 +129,27 @@ class Mailable implements MailableContract, Renderable
      */
     public function send(MailerContract $mailer)
     {
-        Container::getInstance()->call([$this, 'build']);
+        $currentLocale = $mailer->translator ? $mailer->translator->getLocale() : null;
 
-        $mailer->send($this->buildView(), $this->buildViewData(), function ($message) {
-            $this->buildFrom($message)
-                 ->buildRecipients($message)
-                 ->buildSubject($message)
-                 ->runCallbacks($message)
-                 ->buildAttachments($message);
-        });
+        try {
+            if ($mailer->translator && $this->locale) {
+                $mailer->translator->setLocale($this->locale);
+            }
+
+            Container::getInstance()->call([$this, 'build']);
+
+            $mailer->send($this->buildView(), $this->buildViewData(), function ($message) {
+                $this->buildFrom($message)
+                    ->buildRecipients($message)
+                    ->buildSubject($message)
+                    ->runCallbacks($message)
+                    ->buildAttachments($message);
+            });
+        } finally {
+            if ($mailer->translator) {
+                $mailer->translator->setLocale($currentLocale);
+            }
+        }
```

Comment:
Would you have an interest in offering a `ChangesLocale` concerns trait that can be re-purposed in user-land?

```php
trait ChangesLocale
{
    /**
     * @param \Illuminate\Contracts\Translation\Translator $translator
     * @param string                                       $locale
     * @param callable                                     $callback
     *
     * @return mixed
     */
    public function asLocale($translator, $locale, callable $callback)
    {
        if ($translator && $locale) {
            try {
                $currentLocale = $translator->getLocale();

                $translator->setLocale($locale);

                return $callback($locale);
            } finally {
                $translator->setLocale($currentLocale);
            }
        } else {
            return $callback($locale);
        }
    }
}
```

Then `send()` simplifies to:

```php
public function send(MailerContract $mailer)
{
    $this->asLocale($mailer->translator, $this->locale, function () use ($mailer) {
        Container::getInstance()->call([$this, 'build']);

        $mailer->send($this->buildView(), $this->buildViewData(), function ($message) {
            $this->buildFrom($message)
                ->buildRecipients($message)
                ->buildSubject($message)
                ->runCallbacks($message)
                ->buildAttachments($message);
        });
    });
}
```

The trait I've been wrapping `Mailer@send()` with for awhile is much simpler when assuming a translator is bound to the app container:

```php
trait ChangesLocale
{
    public function asLocale($locale, callable $callback)
    {
        try {
            $currentLocale = app()->getLocale();

            if ($locale) {
                app()->setLocale($locale);
            }

            return $callback($locale);
        } finally {
            if ($locale) {
                app()->setLocale($currentLocale);
            }
        }
    }
}
```